### ISSUE TITLE
Japanese event category slugs cause log warning

### ISSUE THREAD

**Issue by kadamwhite** [the-events-calendar/the-events-calendar-category-colors#139]
I noticed today some new log warnings,

```
[09-Nov-2022 15:56:58 UTC] PHP Warning:  Undefined array key "%E3%82%A4%E3%83%99%E3%83%B3%E3%83%88-text" in /plugins/the-events-calendar-category-colors/src/views/category.css.php on line 58
[09-Nov-2022 15:56:58 UTC] PHP Warning:  Undefined array key "%E3%82%A4%E3%83%99%E3%83%B3%E3%83%88-text" in /plugins/the-events-calendar-category-colors/src/views/category.css.php on line 59
[09-Nov-2022 15:56:58 UTC] PHP Warning:  Undefined array key "%E3%82%A4%E3%83%99%E3%83%B3%E3%83%88-background" in /plugins/the-events-calendar-category-colors/src/views/category.css.php on line 75
[09-Nov-2022 15:56:58 UTC] PHP Warning:  Undefined array key "%E3%82%A4%E3%83%99%E3%83%B3%E3%83%88-border" in /plugins/the-events-calendar-category-colors/src/views/category.css.php on line 78
[09-Nov-2022 15:56:58 UTC] PHP Warning:  Undefined array key "%E3%82%A4%E3%83%99%E3%83%B3%E3%83%88-text" in /plugins/the-events-calendar-category-colors/src/views/category.css.php on line 82
[09-Nov-2022 15:56:58 UTC] PHP Warning:  Undefined array key "%E3%82%A4%E3%83%99%E3%83%B3%E3%83%88-text" in /plugins/the-events-calendar-category-colors/src/views/category.css.php on line 83
[09-Nov-2022 15:56:58 UTC] PHP Warning:  Undefined array key "%E3%82%A4%E3%83%99%E3%83%B3%E3%83%88-background" in /plugins/the-events-calendar-category-colors/src/Category_Colors/CSS/Pro.php on line 112
```

There is a category on this site with its label and slug in Japanese (reads "event"), 
<img width="474" alt="image" src="https://user-images.githubusercontent.com/442115/200882336-9246c30c-a88e-46e5-8968-38a7824e6f7d.png">
That katakana slug gets encoded as `'%E3%82%A4%E3%83%99%E3%83%B3%E3%83%88'`, which isn't matching properly when trying to interpolate it into the array keys.

I haven't dug in deeper to propose a solution, apologies, but wanted to flag the issue. We're instructing editors to use ascii characters in labels for now.

**Comment by afragen** [comment]
Multilingual support has always been touchy. Take a look at https://wordpress.org/support/topic/wpml-support-67/ maybe it'll help.

**Comment by afragen** [comment]
Though this looks like an issue with how core does something here.

**Comment by afragen** [comment]
I'm fairly certain that any modifications I make to a term are only safe in Arabic letters. I have no idea how to do manage this with Japanese, etc.

**Comment by kadamwhite** [comment]
https://github.com/afragen/the-events-calendar-category-colors/pull/137 would mitigate this for now, I am not sure whether colors work properly with non-ascii slugs but the log warnings are the biggest issue for us right now.

**Comment by afragen** [comment]
I've merged #137, let me know if it also fixes this issue.
